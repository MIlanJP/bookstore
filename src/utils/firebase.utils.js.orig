import firebase from 'firebase';
import firebase1 from 'firebase/app';

import admin from 'firebase-admin'

const firebaseConfig = {
    apiKey: "AIzaSyCF8I46ftSKLpqRBZ1xPIWZXbkA4I2zK04",
    authDomain: "bookstore-5636e.firebaseapp.com",
    databaseURL: "https://bookstore-5636e.firebaseio.com",
    projectId: "bookstore-5636e",
    storageBucket: "bookstore-5636e.appspot.com",
    messagingSenderId: "540723071433",
    appId: "1:540723071433:web:f2065f00acaf7ed9e24018",
    measurementId: "G-NWV2SXZW3D"
  };

  firebase.initializeApp(firebaseConfig)

  export const auth=firebase.auth();
  export const fireStore=firebase.firestore();
  const provider=new firebase.auth.GoogleAuthProvider();
  provider.setCustomParameters({prompt:'select_account'})
  // export const createUserProfileDocument=async(userAuth,additionalData)=>{
  //   // if(userAuth==='') return 
  //   const userRef=fireStore.doc('users/PPuI3rXmthxyGDqGOmML')
  //   const snapshot=await userRef.get()
  //   console.log(JSON.s snapshot)
  //    fireStore.collection('users').get().then( 
  //     (data)=>{
  //       // console.log(data.docs)
   
  //     }
  //   )
  // }
  export const signInWithGoogle=()=>{auth.signInWithPopup(provider)}
export default firebase;

