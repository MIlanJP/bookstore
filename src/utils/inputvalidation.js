
class Validation {
    vallidateName(name) {
      return /^[A-Z]+[a-z1-9]*$/.test(name) && name.length > 4;
    }
    
  }
  
  export default new Validation();