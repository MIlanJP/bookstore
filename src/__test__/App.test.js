import React from 'react'
import {BrowserRouter} from 'react-router-dom'
// We're using our own custom render function and not RTL's render
// our custom utils also re-export everything from RTL
// so we can import fireEvent and screen here as well
import { render, fireEvent, screen } from '../utils/testUtils'
import Profile from '../components/profile'

const userData={
  name: '',
  email: '',
  booksInCart: 0,
  itemsList: [],
}

describe("render profile page",()=>{
  it('Renders the connected app with initialState', () => {
 const {asFragment}= render(<BrowserRouter> <Profile  user="Milan"  userData={userData}  /></BrowserRouter>, { initialState: { user: 'Redux User',userData ,cardquantity:0} })
    expect(screen.getByText("Bookstore")).toBeInTheDocument()
  })

  it("Match SnapShot for profile page",()=>{
 const {asFragment}= render(<BrowserRouter> <Profile  user="Milan"  userData={userData}  /></BrowserRouter>, { initialState: { user: 'Redux User',userData ,cardquantity:0} })
    expect(asFragment()).toMatchSnapshot();
  })

  
  it("Match SnapShot for profile page",()=>{
    const {getByTestId}= render(<BrowserRouter> <Profile  user="Milan"  userData={userData}  /></BrowserRouter>, { initialState: { user: 'Redux User',userData ,cardquantity:0} })
      //  expect(asFragment()).toMatchSnapshot();
     })

})

