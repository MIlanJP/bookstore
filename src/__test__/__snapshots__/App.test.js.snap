// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render profile page Match SnapShot for profile page 1`] = `
<DocumentFragment>
   
  <div
    data-test="profilePage"
  >
    <div
      class="makeStyles-grow-12"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBarColor-22 mui-fixed MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <button
            aria-label="open drawer"
            class="MuiButtonBase-root MuiIconButton-root makeStyles-bookIcon-13 MuiIconButton-colorInherit MuiIconButton-edgeStart"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-book-14"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 11.55C9.64 9.35 6.48 8 3 8v11c3.48 0 6.64 1.35 9 3.55 2.36-2.19 5.52-3.55 9-3.55V8c-3.48 0-6.64 1.35-9 3.55zM12 8c1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3 1.34 3 3 3z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <h6
            class="MuiTypography-root makeStyles-title-15 MuiTypography-h6 MuiTypography-noWrap"
          >
            Bookstore
          </h6>
          <div
            class="makeStyles-search-16"
          >
            <div
              class="makeStyles-searchIcon-17"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <div
              class="MuiInputBase-root makeStyles-inputRoot-18"
            >
              <input
                aria-label="search"
                class="MuiInputBase-input makeStyles-inputInput-19"
                placeholder="Searchâ€¦"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="makeStyles-grow-12"
          />
          <div
            class="makeStyles-sectionDesktop-20"
          >
            <button
              aria-label="show 0 Items in Carts"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <span
                  class="MuiBadge-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
                    />
                  </svg>
                  <span
                    class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary MuiBadge-invisible"
                  />
                </span>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              aria-controls="primary-search-account-menu"
              aria-haspopup="true"
              aria-label="account of current user"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeEnd"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="makeStyles-sectionMobile-21"
          >
            <button
              aria-controls="primary-search-account-menu-mobile"
              aria-haspopup="true"
              aria-label="show more"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </header>
    </div>
    <div>
      Milan profile
    </div>
  </div>
</DocumentFragment>
`;
